src/backend/core/mda/outbound.py

```diff
@@ -64,6 +65,37 @@ def prepare_outbound_message(
     # Generate a MIME id
     message.mime_id = message.generate_mime_id()
 
+    # Insert the validated signature
+    validated_signature = mailbox_sender.get_validated_signature(
+        message.signature.id if message.signature else None, user
+    )
+    if message.signature != validated_signature:
+        message.signature = validated_signature
+        message.save(update_fields=["signature"])
+    if message.signature:
+        try:
+            signatures = message.signature.render_template(
+                mailbox=mailbox_sender, user=user
+            )
+            if signatures:
+                text_body = (
+                    text_body + "\n" + signatures["text_body"]
+                    if text_body
+                    else signatures["text_body"]
+                )
+                html_body = (
+                    html_body + "<p>" + signatures["html_body"] + "</p>"
```

Comment:
Do we need to wrap in a <p> ? Won't this produce invalid markup with some tags in html_body ?